pyfr/writers/vtk.py

```diff
@@ -9,18 +9,56 @@
 from pyfr.shapes import BaseShape
 from pyfr.util import memoize, subclass_where
 from pyfr.writers import BaseWriter
+from pyfr.writers.nodemaps import VTKHONodeMaps
 
 
 class VTKWriter(BaseWriter):
     # Supported file types and extensions
     name = 'vtk'
     extn = ['.vtu', '.pvtu']
 
+    vtk_types_ho = dict(tri=69, quad=70, tet=71, pri=73, hex=72)
+
     def __init__(self, args):
         super().__init__(args)
 
         self.dtype = np.dtype(args.precision).type
-        self.divisor = args.divisor or self.cfg.getint('solver', 'order')
+
+        # Divisor for each type element
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Choose whether to output subdivided cells or high order VTK cells
+        # If -k is given by the user then use high-order VTK cells as output
+        # with order equal to the solution order or to the one provided by
+        # the user
+        # Else if neither -o nor -d are found in the input then use high-order
+        # VTK cells with order equal to the simulation order
+        # Else use cell subdivision
+        if args.order:
+            self.ho_output = True
+            self.divisor = args.order
+        elif args.divisor is None:
+            self.ho_output = True
+            self.divisor = self.cfg.getint('solver', 'order')
+        else:
+            self.ho_output = False
+            self.divisor = args.divisor
+
+        # If using high-order VTK output mode
+        # set the number of sub-divisions of pyr
+        # elements to divisor + 2
+        if self.ho_output:
```

Comment:
Can merge with the previous if statement

Reply:
Didn't get this one. (see if the new modifications work for you). If not, we have if -> elif -> else and I would need to include the special `pyr` case in the if and elif.?

Reply:
Should now be able to merge the blocks